core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1335,41 +1318,32 @@ class ReassignedPartitionsIsrChangeListener(controller: KafkaController, topic:
     }
   }
 
-  /**
-   * @throws Exception
-   *             On any error.
-   */
-  @throws(classOf[Exception])
-  def handleDataDeleted(dataPath: String) {
-  }
+  @throws[Exception]
+  def doHandleDataDeleted(dataPath: String) {}
+
 }
 
 /**
  * Called when leader intimates of isr change
  *
  * @param controller
  */
-class IsrChangeNotificationListener(controller: KafkaController) extends IZkChildListener with Logging {
+class IsrChangeNotificationListener(protected val controller: KafkaController) extends ControllerZkChildListener {
 
-  override def handleChildChange(parentPath: String, currentChildren: util.List[String]): Unit = {
-    import scala.collection.JavaConverters._
-    // Avoid continue to exectue handleChildChange after controller is not active because of zkclient's callback order
-    if(!controller.isActive()) {
-      return
-    }
+  protected def logName = "IsrChangeNotificationListener"
 
+  def doHandleChildChange(parentPath: String, currentChildren: Seq[String]): Unit = {
```

Comment:
Hmm, to add that, we'd want to verify that the method can actually throw a checked exception. Where can that happen, I don't see it.